.gitlab-ci.yml

```diff
@@ -117,25 +118,64 @@ stages:
     CHAIN_URL:                     "wss://${CHAIN_NAME}-testing-rpc.parity-build.parity.io"
   script:
     - yarn --immutable
-    - yarn test:latest-e2e-tests --chain $CHAIN_NAME --local $CHAIN_URL
+    - echo "Tests for ${CHAIN_URL}"
+    - yarn test:latest-e2e-tests --chain ${CHAIN_NAME} --local ${CHAIN_URL}
 
-e2e-westend:
-  <<:                              *e2e-template
+# e2e-westend:
+#   <<:                              *e2e-template
+#   variables:
+#     CI_IMAGE:                      $NODE_IMAGE
+#     CHAIN_NAME:                    "westend"
+#     CHAIN_URL:                     "wss://${CHAIN_NAME}-testing-rpc.parity-build.parity.io"
+
+# e2e-kusama:
+#   <<:                              *e2e-template
+#   variables:
+#     CI_IMAGE:                      $NODE_IMAGE
+#     CHAIN_NAME:                    "kusama"
+#     CHAIN_URL:                     "wss://${CHAIN_NAME}-testing-rpc.parity-build.parity.io"
+
+# e2e-polkadot:
+#   <<:                              *e2e-template
+#   variables:
+#     CI_IMAGE:                      $NODE_IMAGE
+#     CHAIN_NAME:                    "polkadot"
+#     CHAIN_URL:                     "wss://${CHAIN_NAME}-testing-rpc.parity-build.parity.io"
+
+.bench-template:                   &bench-template
+  stage:                           benchmark
+  <<:                              *kubernetes-env
+  <<:                              *collect-artifacts
   variables:
-    CI_IMAGE:                      $NODE_IMAGE
+    CI_IMAGE:                      $BENCHMARK_IMAGE
+    CHAIN_NAME:                    ""
+    CHAIN_URL:                     "wss://${CHAIN_NAME}-testing-rpc.parity-build.parity.io"
+  script:
+    - yarn --immutable
+    - echo "Benchmarks for ${CHAIN_URL}"
+    - mkdir artifacts
+    - yarn bench --ws-url ${CHAIN_URL} > artifacts/${CHAIN_NAME}_benches.txt
+
+bench-westend:
+  <<:                              *bench-template
+  variables:
+    CI_IMAGE:                      $BENCHMARK_IMAGE
     CHAIN_NAME:                    "westend"
+    CHAIN_URL:                     "wss://${CHAIN_NAME}-testing-rpc.parity-build.parity.io"
 
-e2e-kusama:
-  <<:                              *e2e-template
+bench-kusama:
```

Comment:
So for our benches, I might have mentioned it earlier, but we are only benchmarking against Polkadot if that is alright.

Reply:
Ah, okay, I was about to ask about pallets that don't exist in other chains :)